mobile/src/main/java/org/openhab/habdroid/ui/activity/AbstractWebViewFragment.kt

```diff
@@ -91,8 +93,10 @@ abstract class AbstractWebViewFragment : Fragment(), ConnectionFactory.UpdateLis
                 .putExtra(MainActivity.EXTRA_SERVER_ID, context.getPrefs().getActiveServerId())
                 .setAction(shortcutAction)
 
-            webView?.url?.toHttpUrlOrNull()?.let {
-                intent.putExtra(MainActivity.EXTRA_SUBPAGE, it.toRelativeUrl())
+            Handler(Looper.getMainLooper()).post {
```

Comment:
Should use coroutines here instead of Handler.
(And BTW, I think it would be good to replace the various `GlobalScope.launch` calls in this file by making the fragment a scope by itself, and moving the `launch` calls to the respective JS interfaces.)